shell/components/firebase-upload.js

```diff
@@ -0,0 +1,62 @@
+import Xen from './xen/xen.js';
+import Firebase from './firebase.js';
+
+const {firebase, storage} = Firebase;
+
+const template = `
+  <style>
+    ::slotted(*) {
+      cursor: pointer;
+    }
+    input[type="file"] {
+      display: none;
+    }
+  </style>
+  <label>
+    <slot></slot>
+    <input type="file" accept="{{accept}}" multiple="{{multiple}}" on-change="_onFilesChanged">
+  </label>
+`;
+
+class FirebaseUpload extends Xen.Base {
+  get template() {
+    return template;
+  }
+  static get observedAttributes() {
+    return ['multiple', 'accept'];
+  }
+  _render({accept, multiple}) {
+    return {
+      accept,
+      multiple
+    };
+  }
+  _onFilesChanged(e) {
+    const input = e.currentTarget;
+    const file = input.files[0];
+    const random = Math.floor((Math.random()+1)*1e8);
+    const path = `files/${random}`;
+    console.log('uploading', file.name, 'as', path);
+    this._uploadFile(file, path);
+  }
+  _uploadFile(file, uploadPath) {
+    const imageRef = storage.ref().child(uploadPath);
+    const next = () => {
+    };
+    const error = error => {
+      console.error('Error uploading', error);
```

Comment:
Log the file name and upload path since it may not be in the error?

Reply:
I have no idea what those errors looks like unfortunately, hopefully it's documented.

We also need to send events for errors and progress.